### DISCUSSION TITLE
Playing Music when switching scene

### DISCUSSION THREAD

**Discussion by GamesStudioTwo** [axmolengine/axmol#2094] [General]
HI guys hope your all doing well. I'm changing scene and a I had music playing in background is there a way to change scene and keep the music going or do I have to play again. I know changing scene flushes all resources witch is very nice by the way.
but I would_ like the music to continue when changing is it posable ?.
Many thanks in advance.

**Comment by paulocoutinhox** [comment]
Hi,

You can use:

```
AudioEngine::preload("sounds/music.mp3");
AudioEngine::play2d("sounds/music.mp3", true);
```

It will play sound in background for your game independent of scene. It never stop.

Thanks.

**Comment by GamesStudioTwo** [comment]
I think there's a bug in my class because when I switch from a different scene it's fine but the new Map class scene seems to stop it.
don't know why but I will remake it as its empty anyway.

**Reply by paulocoutinhox** [reply]
It could be something in your code that is interfering with this, because in my game it works without problems. The other audios must be played without the second Boolean parameter.

**Reply by GamesStudioTwo** [reply]
solved it. it's was in my other class I told audio Engine to shutdown when the destructor was called in that class from this one.
ill  remove it from destructor LOL.
Thanks' for the help buddy.

Edit: lol its even Worse it's 3 classis deep. I forgot I made a hole Sound effect class OMG and I'm not using it what a noob I am.
